Libraries/Blob/URL.js

```diff
@@ -151,6 +151,21 @@ export class URL {
       }
       this._url = `${baseUrl}${url}`;
     }
+
+    let queryStringStart = this._url.indexOf('?');
+    if (queryStringStart < 0) {
+      return;
+    }
+    let queryString = this._url.substring(queryStringStart + 1);
+
+    this._url = this._url.substr(0, queryStringStart);
+    let searchParams = queryString.split('&').reduce((agg, pair) => {
+      let [key, value] = pair.split('=');
```

Comment:
I can imagine a couple common issues with this.

- Empty query string segments: `/a=b&`
- Empty query parameter values: `/a`
- Presence of `=` character in query parameter values: `/a=b=c`

Reply:
I've added handling and test cases for these three and verified that the behavior mimics what I see in the browser.